session3/gulp-tutorial/gulpfile.js

```diff
@@ -0,0 +1,52 @@
+const gulp = require('gulp');
+const imagemin = require('gulp-imagemin');
+const uglify = require('gulp-uglify');
+const sass = require('gulp-sass');
+const concat = require('gulp-concat');
+const browserSync = require('browser-sync').create();
+
+
+gulp.task('message', function () {
+    return console.log('gulp is running');
+});
+
+gulp.task('default',['browserSync','message','copyHTML', 'imagemin', 'sass', 'scripts','watch']);
+
+gulp.task('copyHTML',function () {
+    gulp.src('src/*.html').pipe(gulp.dest('dist'));

```

Comment:
gulp tasks need a promise in order to run tasks in series or parallel. Make sure you are using return statement, or call the gulp task callback after completing the task.